scripts/genRef_1.R

```diff
@@ -0,0 +1,65 @@
+## Reference matrix type 1 (rows = genenames, cols = celltypes) generation script
+##
+## @zgr2788
+
+suppressMessages(library(Seurat))
+suppressMessages(library(sparseMatrixStats))
+
+#Read data
+args <- commandArgs(trailingOnly = TRUE)
+filename <- args[1]
+transform <- args[2]
+C_0 <- readRDS(filename)
+#filename <- sub("Input/Cell_splits", "Input/References", filename)
+
+#normalize
+
+#Write sc-counts matrix as separate object to save memory later on
+C_counts <- C_0@assays$RNA@counts
+C_metadata <- C_0@meta.data
+
+#Initialize references
+C_1 <- data.frame(row.names = rownames(C_0@assays$RNA@counts))
+
+#Transform C matrix
+switch(transform,
+
+  "none" = {
+    message('Warning: No transformation applied')
+  },
+
+  "log" = {
+    C_0@assays$RNA@counts <- log1p(C_0@assays$RNA@counts)
+  },
+
+  "sqrt" = {
+    C_0@assays$RNA@counts <- sqrt(C_0@assays$RNA@counts)
+  },
+
+  "vst" = { #DO NOT USE
+    #Not recommended for low memory, uses dense matrix
+    #vst requires int, convert first
+    C_0@assays$RNA@counts <- round(C_0@assays$RNA@counts)
+
+    C_0@assays$RNA@counts <- DESeq2::varianceStabilizingTransformation(C_0@assays$RNA@counts)
+  }
+
+)
```

Comment:
R scripts can benefit from a style formatting. Example:
```suggestion
switch(transform,
  "none" = {
    message('Warning: No transformation applied')
  },
  "log" = {
    C_0@assays$RNA@counts <- log1p(C_0@assays$RNA@counts)
  },
  "sqrt" = {
    C_0@assays$RNA@counts <- sqrt(C_0@assays$RNA@counts)
  },
  "vst" = {  # do not use
    message('Warning: varianceStabilizingTransformation requires high memory and is not recommended.')
    C_0@assays$RNA@counts <- round(C_0@assays$RNA@counts)
    C_0@assays$RNA@counts <- DESeq2::varianceStabilizingTransformation(C_0@assays$RNA@counts)
  }
)

```